core/src/main/java/org/mule/routing/correlation/EventCorrelator.java

```diff
@@ -67,6 +66,8 @@
 
     private static final long ONE_DAY_IN_MILLI = 1000 * 60 * 60 * 24;
 
+    private static int processedMessages = 0;
```

Comment:
If this is static you will have concurrency problems. I think it will not affect the behaviour but the splitted message could have incoherent suffixes.